pkg/storer/internal/cache/cache.go

```diff
@@ -295,69 +265,54 @@ func (c *Cache) Putter(store internal.Storage) storage.Putter {
 			return fmt.Errorf("failed creating batch: %w", err)
 		}
 
-		state := &cacheState{}
-		err = store.IndexStore().Get(state)
+		newEntry.AccessTimestamp = now().UnixNano()
+		err = batch.Put(newEntry)
 		if err != nil {
-			return fmt.Errorf("failed reading cache state: %w", err)
+			return fmt.Errorf("failed adding cache entry: %w", err)
 		}
 
-		if state.Count != 0 {
-			// if we are here, this is a new chunk which should be added. Add it to
-			// the end.
-			err = c.pushBack(
+		err = batch.Put(&cacheOrderIndex{
+			Address:         newEntry.Address,
+			AccessTimestamp: newEntry.AccessTimestamp,
+		})
+		if err != nil {
+			return fmt.Errorf("failed adding cache order index: %w", err)
+		}
+
+		addSize := true
+		if c.size.Load() == int64(c.capacity) {
+			addSize = false
+			err := removeOldest(
 				ctx,
-				state,
-				newEntry,
-				chunk,
 				store.IndexStore(),
 				batch,
 				store.ChunkStore(),
+				1,
```

Comment:
here maybe we can remove at least 1,000~50,000 items so we do not have to remove oldest entries every time.